mono/metadata/gc.c

```diff
@@ -681,6 +682,31 @@ ves_icall_System_GCHandle_GetTarget (guint32 handle)
 	return mono_gchandle_get_target (handle);
 }
 
+static gboolean
+is_object_pinnable (MonoObject *obj)
```

Comment:
This function doesn't make sense.

Why is it checking for MONO_TYPE_CLASS?
Why is it giving special treatment to char and boolean?
Why it is giving special treatment to ARRAY and SZARRAY in the array case?

The tests need to cover the positive tests as well.